pkg/capacityscheduling/elasticquota.go

```diff
@@ -88,14 +88,26 @@ func (e *ElasticQuotaInfo) unreserveResource(request framework.Resource) {
 }
 
 func (e *ElasticQuotaInfo) usedOverMinWith(podRequest *framework.Resource) bool {
+	// "ElasticQuotaInfo doesn't have Min" means used values exceeded min(0)
+	if e.Min == nil {
```

Comment:
if Min/Max/Used can be nil, maybe we should use a concrete type instead of pointer in ElasticQuotaInfo, to get rid of nil check burden.

@denkensk was there any particular reason making them pointers in the beginning?

Reply:
It's been so long that I can't remember exactly why. 

Maybe We will often call methods in this package https://github.com/kubernetes/kubernetes/blob/5d9d06aa4749ef9ad73f94fef0ef2e56719f9190/pkg/scheduler/framework/types.go#L432; And need to pass a pointer of the struct as an argument to function argument. So I store these as pointers for convenience.

Reply:
Thanks for your comments!

> if Min/Max/Used can be nil, maybe we should use a concrete type instead of pointer in ElasticQuotaInfo, to get rid of nil check burden.

@Huang-Wei 
Do you mean like this?
```
type ElasticQuotaInfo struct {
	Namespace string
	pods      sets.String
	Min       framework.Resource
	Max       framework.Resource
	Used      framework.Resource
}

```

In my opinion, Min / Max / Used should be used as pointer.
Firstly, as @denkensk said, it seems kubernetes's framework package recommends us to use pointer.
The return value of NewResource is pointer as well.
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/types.go#L444-L448

Secondly, if we change Min/Max/Used as concrete type, we would have to fix many parts in capacityScheduling.
This bug is so small, so we don't need to change many parts, just only for this reason.
This problem should be fixed in each useOver functions.

From these reasons, I think Min / Max / Used should be used as pointer.
But you two are so familiar with scheduler-plugin, so any advice would be appreciated.
Thanks,

Reply:
> This bug is so small, so we don't need to change many parts, just only for this reason.

Makes sense to me.